packages/react-server/core/ClientController.js

```diff
@@ -34,12 +34,13 @@ Q.onerror = (err) => {
 	logger.error("Unhandled exception in Q promise", err);
 }
 
-var CURRENT_STATE_FRAME = 0;
+var SESSION_START_PREFIX = (new Date()).getTime() + '_';
 var NEXT_STATE_FRAME = 0;
+var CURRENT_STATE_FRAME = SESSION_START_PREFIX + NEXT_STATE_FRAME;
```

Comment:
I don't like having two places where frame identifiers are created (inline here and in `pushFrame`).

The _initial_ frame isn't used (new frames are always obtained via `pushFrame()`), so this could just be declared without assignment (left `undefined`) or made explicitly `null`:

```javascript
var CURRENT_STATE_FRAME; // No frame until `pushFrame()` is called.
```

Could also initialize with a valid (though unused) identifier while still avoiding duplication by just calling `pushFrame()` and burning the first frame:

```javascript
var CURRENT_STATE_FRAME = pushFrame();
```

`pushFrame` gets hoisted and the enclosed variables are all defined before `CURRENT_STATE_FRAME` so this wouldn't require re-ordering anything.  Kind of weird to always burn a frame identifier, though.